ihmc-perception/src/main/java/us/ihmc/perception/gpuHeightMap/RapidHeightMapExtractorCUDA.java

```diff
@@ -297,28 +308,42 @@ public void update(RigidBodyTransform sensorToWorldTransform, RigidBodyTransform
       // Run the cropping kernel
       croppingKernel.withPointer(globalHeightMapImage.data()).withLong(globalHeightMapImage.step());
       croppingKernel.withPointer(sensorCroppedHeightMapImage.data()).withLong(sensorCroppedHeightMapImage.step());
+      croppingKernel.withPointer(terrainHeightMapImage.data()).withLong(terrainHeightMapImage.step());
       croppingKernel.withPointer(parametersDevicePointer);
       croppingKernel.withInt(heightMapParameters.getCropWindowSize());
+      croppingKernel.withInt(terrainHeightMapImage.rows());
       error = cudaStreamSynchronize(stream);
       CUDATools.checkCUDAError(error);
 
       croppingKernel.run(stream, croppingKernelGridDim, blockSize, 0);
       error = cudaStreamSynchronize(stream);
       CUDATools.checkCUDAError(error);
 
-      snappedFootstepsExtractor.update(globalHeightMapImage, sensorOrigin, globalCenterIndex, cropCenterIndex);
+      snappedFootstepsExtractor.update(globalHeightMapImage, sensorOrigin, globalCenterIndex, terrainCenterIndex);
 
       //Update the terrain map data with the new results
       terrainMapData.setSensorOrigin(groundToWorldTransform.getTranslationX(), groundToWorldTransform.getTranslationY());
 
-      Mat finalCroppedHeightMap = new Mat();  // Assuming the height map is 201x201
-      sensorCroppedHeightMapImage.download(finalCroppedHeightMap);  // Download the image from the GPU to the Mat object
+      Mat temp = new Mat();
+      terrainHeightMapImage.download(temp);
+//      PerceptionDebugTools.printMat("s", temp, 5);
+
+      Mat finalCroppedHeightMap = new Mat();
+      if (heightMapParameters.getEnableVerticalFilter())
+      {
+         GpuMat verticalFilteredMap = verticalSurfacesExtractor.update(terrainHeightMapImage);
+         verticalFilteredMap.download(finalCroppedHeightMap);  // Download the image from the GPU to the Mat object
```

Comment:
Close the GpuMat after use
```suggestion
         verticalFilteredMap.download(finalCroppedHeightMap);  // Download the image from the GPU to the Mat object
         verticalFilteredMap.close();
```

Reply:
This suggestion's formatting looks weird.. I wouldn't apply it directly haha